src/fdcache_stat.cpp

```diff
@@ -193,6 +193,47 @@ bool CacheFileStat::SetPath(const char* tpath, bool is_open)
     return Open();
 }
 
+// [NOTE]
+// There is no need to check whether the file is open because using rename().
+//
+bool CacheFileStat::OverWriteFile(const std::string& strall) const
+{
+    // make temporary file path(in same cache directory)
+    std::string sfile_path;
+    if(0 != CacheFileStat::MakeCacheFileStatPath(path.c_str(), sfile_path, true)){
+        S3FS_PRN_ERR("failed to create cache stat file path(%s)", path.c_str());
+        return false;
+    }
+    std::string strTmpFile = mydirname(sfile_path) + "/.tmpstat.XXXXXX";
+    char *pTempFileName    = strdup(strTmpFile.c_str());
+
+    // open temporary file
+    int tmpfd;
+    if(-1 == (tmpfd = mkstemp(pTempFileName))){     // mode: 0600
```

Comment:
Can we simplify this by calling `mkstemp(strTmpFile.data())` (actually this requires two `std::string`)?  This will requires to C++17 but we plan to do that anyway.  My objection is using manual memory management like `free`.

Reply:
No problem. I fixed it by using `&str[]` to make it compatible with C++14.